Second tutorial: Connect the PIM (#321)
akeneo/pim-api-docs: content/getting-started/connect-the-pim-old/step-2.md

![Permission assignation](/img/getting-started/permission-assignation-my-erp-user.png)
1. Click on the `Save` button.

::: info
_Best practice_  
Do not - under any condition :wink: - re-use a user role that is assigned to real users. Indeed, we highly recommend you to create a dedicated user role for each one of your connectors.  
This way, you can give this user role, exactly the API permissions needed for your connector to work.
:::

## Get your client ID and secret

Next, you will need to generate a client ID and a secret for your connection.

::: info
_Best practice_  
Generate one set of client ID/secret by connector. Indeed, we advise you not to share them between your connectors.  
It will simplify the management of these credentials. For example, in case one of the secret leaks, it won't impact all your connectors at once. :wink:
:::

Depending on your PIM version (v1.7 or v2/v3), there are two methods to generate your client ID and secret.

If you are using a v2 or V3 PIM, [create an API connection](#by-creating-an-API-connection).  
If you are using a v1.7, [launch a command line](#by-command-line-for-v17-pim-only).

### By creating an API connection

> We will create an API connection. We will call it, for the purpose of this tutorial: `ERP`

Reviewer: ```suggestion
We will create an API connection. We will call it `ERP` for the purpose of this tutorial.
```

::: warning

> This feature is only available since the 2.0 version. If you are trying to use the API on a 1.7 PIM, please have a look to the [second method below](#get-your-credentials-via-command-line-).

Reviewer: ```suggestion
This feature is only available since the 2.0 version. If you are trying to use the API on a 1.7 PIM, please take a look at the [second method below](#get-your-credentials-via-command-line-).
```

:::

1. Click on the `System` menu.
1. Click on the `API connections` submenu.
1. Click on the `Create` button.
![API connections empty screen](/img/getting-started/api-connections-empty-screen.png)
The following popin should be then displayed.
![API connection creation popin](/img/getting-started/api-connection-creation-popin.png)
1. Enter `ERP` in the label field and click on the `Save` button.  

> You API connection appears in the API connections table and should look like this:

Reviewer: ```suggestion
Your API connection appears in the API connection table and should look like this:
```

![My very first API connection](/img/getting-started/my-erp-connection.png)

And that's it! :tada:

::: panel-link Now that you generated your credentials, let's move on to [The connector configuration](/getting-started/connect-the-pim-old/step-3.html)
:::

> ### By command line for v1.7 PIM only

Reviewer: ```suggestion
### Use a command line for the v1.7 PIM only
```

# _Step 2_ | Generate your credentials

In this step, you are going to generate your credentials. :rocket:

Indeed, you will need an API user, a password, a client ID and a secret if you want the connector you just cooked, to work. So let's get those credentials!

::: info
_Best practice_  
We strongly advise you to create **one set of credentials** for each one of your connectors.  

> Indeed, it will ease a lot your migration to the PIM v4, in case you are planning it.

Reviewer 2: ```suggestion
Indeed, it will ease a lot your migration to the PIM v4, in case you are planning for it.
```
